<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>380</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="maximumSize">
   <size>
    <width>640</width>
    <height>380</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Observer Pattern Demo</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLineEdit" name="lineEdit">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>350</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_3">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>350</y>
      <width>141</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Notify Observers</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>20</y>
      <width>521</width>
      <height>91</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>FreeSans</family>
     </font>
    </property>
    <property name="title">
     <string>Publisher</string>
    </property>
    <widget class="QPushButton" name="pushButton_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>171</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>FreeSans</family>
       <pointsize>11</pointsize>
       <weight>50</weight>
       <bold>false</bold>
      </font>
     </property>
     <property name="toolTip">
      <string>Connect to MQTT server</string>
     </property>
     <property name="text">
      <string>Connect</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>171</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>FreeSans</family>
      </font>
     </property>
     <property name="text">
      <string>Select GeoJson File</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>130</y>
      <width>521</width>
      <height>151</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>FreeSans</family>
     </font>
    </property>
    <property name="title">
     <string>Subscriber</string>
    </property>
    <widget class="QPushButton" name="pushButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>171</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>FreeSans</family>
      </font>
     </property>
     <property name="text">
      <string>Create new Observer</string>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>41</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>FreeSans</family>
      </font>
     </property>
     <property name="text">
      <string>Name:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_2">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>30</y>
       <width>121</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>20</y>
       <width>301</width>
       <height>121</height>
      </rect>
     </property>
     <property name="title">
      <string>Render Style</string>
     </property>
     <widget class="QRadioButton" name="radioButton_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>112</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>FreeSans</family>
       </font>
      </property>
      <property name="text">
       <string>Text</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="radioButton">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>112</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>FreeSans</family>
       </font>
      </property>
      <property name="text">
       <string>Graphic</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="radioButton_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>112</width>
        <height>23</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>FreeSans</family>
       </font>
      </property>
      <property name="text">
       <string>Stats</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
